compiler/luci/pass/include/luci/ModulePass.h

```diff
@@ -22,16 +22,28 @@
 
 #include <luci/IR/Module.h>
 
+#include <cassert>
+
 namespace luci
 {
 
-class Pass : public logo::Pass
+class ModulePass
 {
 public:
   // Run module pass and return false if there was nothing changed
   virtual bool run(luci::Module *) = 0;
 };
 
+class Pass : public logo::Pass, public ModulePass
+{
+public:
+  // NOTE adding dummy run() to make compiler happy with "-Werror=overloaded-virtual="
+  // clang-format off
+  bool run(loco::Graph *) override { assert(false); return false; }
+  bool run(luci::Module *) override { assert(false); return false; }
```

Comment:
How about throwing a runtime error instead of using assert?
Please note that the error message might not match what you intended.

```suggestion
  bool run(loco::Graph *) override { throw std::runtime_error("Pass::run(loco::Graph *) should never be called"); }
  bool run(luci::Module *) override { throw std::runtime_error("Pass::run(luci::Module *) should be overridden by subclasses"); }
```

Reply:
I'll revise to use throw with `Must inherit` message :)